starknet_py/proxy/contract_abi_resolver.py

```diff
@@ -0,0 +1,150 @@
+import warnings
+from enum import Enum
+from typing import List, TypedDict, Tuple, Optional
+
+from starknet_py.net.client import Client
+from starknet_py.net.client_errors import ContractNotFoundError, ClientError
+from starknet_py.net.client_models import Abi, DeclaredContract
+from starknet_py.net.models import Address
+from starknet_py.proxy.proxy_check import (
+    OpenZeppelinProxyCheck,
+    ArgentProxyCheck,
+    ProxyCheck,
+)
+
+
+class ProxyConfig(TypedDict, total=False):
+    """
+    Proxy resolving configuration
+    """
+
+    proxy_checks: List[ProxyCheck]
+    """
+    List of classes implementing :class:`starknet_py.proxy.proxy_check.ProxyCheck` ABC,
+    that will be used for checking if contract at the address is a proxy contract.
+    """
+
+
+def prepare_proxy_config(proxy_config: ProxyConfig) -> ProxyConfig:
+    if "max_steps" in proxy_config:
+        warnings.warn(
+            "ProxyConfig.max_steps is deprecated. Contract.from_address always makes at most 1 step.",
+            category=DeprecationWarning,
+        )
+    proxy_checks = [OpenZeppelinProxyCheck(), ArgentProxyCheck()]
+    return {"proxy_checks": proxy_checks, **proxy_config}
```

Comment:
Should we be always appending default `proxy_checks` to the user provided? What if the user doesn't want to resolve for e.g. OZ proxies?